docs/engine/Life-of-a-Flutter-Frame.md

```diff
@@ -41,6 +40,27 @@ Once a surface is acquired, the [LayerTree][layerTree] is rasterized to the surf
 
 The above process is repeated until the pipeline is empty.
 
+## Warm-up frame
+
+Normally, the Flutter framework begins producing a frame when it receives
+a vsync event from the operating system. However, this may not happen for
+several milliseconds after the app starts (or after a hot reload). To make
+use of the time between when the widget tree is first configured and when
+the engine requests an update, the framework schedules a _warm-up frame_
+using [PlatformDispatcher.scheduleWarmUpFrame][scheduleWarmUpFrame].
+
+A warm-up frame may never actually render (as the engine did not request
+it and therefore does not have a valid context in which to paint), but it
+will cause the framework to go through the steps of building, laying out,
+and painting, which can together take several milliseconds. Thus, when the
+engine requests a real frame, much of the work will already have been
+completed, and the framework can generate the frame with minimal
+additional effort.
```

Comment:
```suggestion
and painting, which allows the first real frame to be generated accumulatively, shortening the delay.
```
Maybe a more concise way?

Reply:
I agree the current text is verbose, but I think it's clearer. (I stole this text from the framework's docs, not tooting my own horn here 😂)